cloud/scope/clients.go

```diff
@@ -79,46 +68,14 @@ func (c *AzureClients) setCredentials(subscriptionID string) error {
 		return err
 	}
 
-	if subscriptionID == "" {
-		subscriptionID = settings.GetSubscriptionID()
-		if subscriptionID == "" {
-			return fmt.Errorf("error creating azure services. subscriptionID is not set in cluster or AZURE_SUBSCRIPTION_ID env var")
-		}
-	}
-
-	c.subscriptionID = subscriptionID
-	c.tenantID = strings.TrimSuffix(settings.Values[auth.TenantID], "\n")
-	c.clientID = strings.TrimSuffix(settings.Values[auth.ClientID], "\n")
-	c.clientSecret = strings.TrimSuffix(settings.Values[auth.ClientSecret], "\n")
-
-	c.environment = settings.Values[auth.EnvironmentName]
-	if c.environment == "" {
-		c.environment = azure.PublicCloud.Name
-	}
-
+	c.EnvironmentSettings = settings
 	c.ResourceManagerEndpoint = settings.Environment.ResourceManagerEndpoint
-	c.ResourceManagerVMDNSSuffix = GetAzureDNSZoneForEnvironment(settings.Environment.Name)
-	settings.Values[auth.SubscriptionID] = subscriptionID
-	settings.Values[auth.TenantID] = c.tenantID
+	c.ResourceManagerVMDNSSuffix = settings.Environment.ResourceManagerVMDNSSuffix
+	c.Values[auth.ClientID] = strings.TrimSuffix(c.Values[auth.ClientID], "\n")
+	c.Values[auth.ClientSecret] = strings.TrimSuffix(c.Values[auth.ClientSecret], "\n")
+	c.Values[auth.SubscriptionID] = strings.TrimSuffix(subscriptionID, "\n")
+	c.Values[auth.TenantID] = strings.TrimSuffix(c.Values[auth.TenantID], "\n")
 
-	c.Authorizer, err = settings.GetAuthorizer()
+	c.Authorizer, err = c.GetAuthorizer()
 	return err
 }
-
-// GetAzureDNSZoneForEnvironment returnes the DNSZone to be used with the
-// cloud environment, the default is the public cloud
-func GetAzureDNSZoneForEnvironment(environmentName string) string {
```

Comment:
does `ResourceManagerVMDNSSuffix` get set to the right values from the `AZURE_ENVIRONMENT` var now?

Reply:
Indeed:

https://github.com/Azure/go-autorest/blob/8f84370b53ea066bda881adc6e07455c58b4b7a6/autorest/azure/environments.go#L98
https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go#L132
https://github.com/Azure/go-autorest/blob/8f84370b53ea066bda881adc6e07455c58b4b7a6/autorest/azure/environments.go#L166
https://github.com/Azure/go-autorest/blob/8f84370b53ea066bda881adc6e07455c58b4b7a6/autorest/azure/environments.go#L200

Reply:
nice